### PR TITLE
Add typing for tmt/steps/report/junit.py

### PR DESCRIPTION
**Pull request by janhavlin**
Blocked until #1319 is merged.

### CODE REVIEW

**Review by optak — COMMENTED** [review]
LGTM

**Review by optak — APPROVED** [review]

File: tmt/steps/report/junit.py [hunk 1]
```diff
@@ -49,19 +62,20 @@ class ReportJUnit(tmt.steps.report.ReportPlugin):
     _keys = ["file"]
 
     @classmethod
-    def options(cls, how=None):
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
         """ Prepare command line options for connect """
-        return [
+        return cast(List[tmt.options.ClickOptionDecoratorType], [
```
Comment (janhavlin) [inline tmt/steps/report/junit.py:67]: We should try to get rid of this `cast()`. PR #1324 has [a proposal](https://github.com/teemtee/tmt/pull/1324#discussion_r926567419) on how to do that, so I'll wait if it gets approved before changing it here also.

Comment (psss) [reply tmt/steps/report/junit.py:67]: I will update this in the same way as for the `html` report.

**Review by psss — APPROVED** [review]
Thanks for preparing the patch. Looks good! Just adjusted the `options()` method.

**Review by thrix — APPROVED** [review]

**Pull request merged**